utils/get_supported_scylla_base_versions.py

```diff
@@ -0,0 +1,131 @@
+#!/usr/bin/env python3
+
+import sys
+import re
+import os
+import click
+
+from pkg_resources import parse_version
+from sdcm.utils.version_utils import is_enterprise
+from sdcm.utils.common import get_s3_scylla_repos_mapping
+
+sys.path.append(os.path.join(os.path.dirname(__file__), ".."))
+
+
+# We support to migrate from specific OOS version to enterprise
+supported_src_oos = {'2021.1': '4.3', '2020.1': '4.0', '2019.1': '3.0'}
+# If new support distro shared repo with others, we need to assign the start support versions. eg: centos8
+start_support_versions = {'centos-8': {'scylla': '4.1', 'enterprise': '2021.1'}}
+
+
+class UpgradeBaseVersion:
+
+    def __init__(self, scylla_version, scylla_repo, linux_distro):
+        if len(linux_distro.split('-')) > 1:
+            self.dist_type, *_, self.dist_version = linux_distro.split('-')
+        else:
+            self.dist_type = linux_distro.split('-')[0]
+            self.dist_version = None
+        self.scylla_repo = scylla_repo
+        self.linux_distro = linux_distro
+        self.product, self.scylla_version = self.get_product_and_version(scylla_version)
+
+    def get_product_and_version(self, scylla_version: str):
+        if scylla_version is None:
+            assert 'unstable/' in self.scylla_repo
+            product, version = self.scylla_repo.split('unstable/')[1].split('/')[0:2]
+            scylla_version = version.replace('branch-', '').replace('enterprise-', '')
+        else:
+            product = 'scylla-enterprise' if is_enterprise(scylla_version) else 'scylla'
+        return product, scylla_version
+
+    def get_start_support_version(self):
+        """
+        We can't detect the support versions for this distro, which shares the repo with others, eg: centos8
+        so we need to assign the start support versions for it.
+        """
+        oos_start_support_version = None
+        ent_start_support_version = None
+        if (self.dist_type not in ['debian', 'ubuntu'] and self.linux_distro != 'centos'):
+            versions_dict = start_support_versions.get(self.linux_distro, None)
+            if versions_dict is None:
+                click.secho(
```

Comment:
Please move out click.echo to the sct.py

Reply:
fixed.